From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: QWRT Build <build@qwrt.local>
Date: Thu, 2 Jul 2026 18:20:00 +0800
Subject: [PATCH] pmdomain: qcom: rpmhpd: allow 32 ARC levels

Radxa's 7.0.11 tree raises the RPMh ARC level table size to 32 entries.
Keep the 6.18 QCOM target aligned so platforms with longer cmd-db level
maps do not fail or truncate power-domain level mapping.

Signed-off-by: QWRT Build <build@qwrt.local>
--- a/drivers/pmdomain/qcom/rpmhpd.c	2026-07-02 16:56:51.127330263 +0800
+++ b/drivers/pmdomain/qcom/rpmhpd.c	2026-07-02 16:56:51.131876160 +0800
@@ -19,7 +19,7 @@
 
 #define domain_to_rpmhpd(domain) container_of(domain, struct rpmhpd, pd)
 
-#define RPMH_ARC_MAX_LEVELS	16
+#define RPMH_ARC_MAX_LEVELS	32
 
 /**
  * struct rpmhpd - top level RPMh power domain resource data structure
@@ -35,7 +35,7 @@
  * @level:		An array of level (vlvl) to corner (hlvl) mappings
  *			derived from cmd-db
  * @level_count:	Number of levels supported by the power domain. max
- *			being 16 (0 - 15)
+ *			being 32 (0 - 31)
  * @enabled:		true if the power domain is enabled
  * @res_name:		Resource name used for cmd-db lookup
  * @addr:		Resource address as looped up using resource name from
